R/randomForest_tidiers.R

```diff
@@ -708,19 +708,43 @@ calc_feature_imp <- function(df,
   # this evaluates select arguments like starts_with
   cols <- dplyr::select_vars(names(df), !!! rlang::quos(...))
 
+  grouped_cols <- grouped_by(df)
+
+  if (any(c(target_col, cols) %in% grouped_cols)) {
+    stop("grouping column is used as variable columns")
+  }
+
+  # remove NA because it's not permitted for randomForest
+  df <- df %>%
+    dplyr::filter(!is.na(!!target_col))
+
+  for (col in cols) {
+    # remove NA from predictor columns
+    df <- df %>%
+      dplyr::filter(!is.na(!!col))
+
+    if(!is.numeric(df[[col]]) && !is.logical(df[[col]])) {
+      # convert data to factor if predictors are not numeric or logical
+      # and limit the number of levels in factor by fct_lump
+      df[[col]] <- forcats::fct_lump(as.factor(df[[col]]), n=predictor_n)
+    }
+  }
+
   # randomForest fails if columns are not clean
   clean_df <- janitor::clean_names(df)
   # this mapping will be used to restore column names
   name_map <- colnames(clean_df)
   names(name_map) <- colnames(df)
 
+  # grouping column should be original column
+  # so the value of grouping columns
+  # should be still the names of grouping columns
```

Comment:
Please keep the explanation you had in the previous version that explains the behavior of clean_names. That info is needed to understand this comment.